transcripts.py

```diff
@@ -191,47 +195,93 @@ def sort(transcript):
 
 
 def sort_key(t):
-	return (t.get('audio_path'), t.get('begin'), t.get('end'), t.get('channel'))
+	return t.get('audio_path'), t.get('begin'), t.get('end'), t.get('channel')
 
 
 def group_key(t):
 	return t.get('audio_path')
 
 
+Interval = typing.NewType('Interval', typing.Tuple[typing.Union[float, int], typing.Union[float, int]])
+
+
 def prune(
-	transcript,
-	align_boundary_words = False,
-	cer = None,
-	wer = None,
-	mer = None,
-	duration = None,
-	gap = None,
-	num_speakers = None,
-	audio_name = None,
-	unk = None,
-	groups = None
+	transcript: Transcript,
+	align_boundary_words: bool = False,
+	cer: typing.Optional[Interval] = None,
+	wer: typing.Optional[Interval] = None,
+	mer: typing.Optional[Interval] = None,
+	duration: typing.Optional[Interval] = None,
+	gap: typing.Optional[Interval] = None,
+	num_speakers: typing.Optional[Interval] = None,
+	allowed_audio_names: typing.Set[str] = None,
+	allowed_unk_count: typing.Optional[Interval] = None,
+	groups: typing.Set = None,
```

Comment:
If `typing.Set` is untyped, pleas use `set` instead. Otherwise, please add a generic argument for the `set` element